install_config/aggregate_logging.adoc

```diff
@@ -3299,3 +3299,118 @@ link:https://github.com/Scribery/tlog[Tlog].
+next DC.
+1. Once all DCs for the cluster have been rolled out, re-enable shard balancing.
+Again from any ES pod in the cluster:
+
+  curl -s --cacert /etc/elasticsearch/secret/admin-ca \
+          --cert /etc/elasticsearch/secret/admin-cert \
+          --key /etc/elasticsearch/secret/admin-key \
+          -XPUT 'https://localhost:9200/_cluster/settings' \
+          -d '{ "transient": { "cluster.routing.allocation.enable" : "all" } }'
+
+
+[[elasticsearch-full-restart]]
+=== Performing an Elasticsearch full cluster restart
+
+A full restart is appropriate to be performed when changing major versions
+of Elasticsearch or other changes which might put data integrity a risk during
+the change process.
+
+[NOTE]
+====
+Any action you do for a ES cluster will need to be repeated for the ops cluster
+if `openshift_logging_use_ops` was configured to be `True`.
+====
+
+[NOTE]
+====
+When making changes to the `logging-es-ops` service use components "es-ops-blocked" and "es-ops" instead in the patch
+====
+
+1. Disable all external communications to the ES cluster while it is down.
+Edit your non-cluster logging service (e.g. `logging-es`, `logging-es-ops`) to no
+longer match the ES pods running:
+
+  oc patch svc/logging-es -p '{"spec":{"selector":{"component":"es-blocked","provider":"openshift"}}}'
+
+1. Prevent shard balancing happening when we are bring down nodes on purpose
+From any ES pod in the cluster:
+
+  curl -s --cacert /etc/elasticsearch/secret/admin-ca \
+          --cert /etc/elasticsearch/secret/admin-cert \
+          --key /etc/elasticsearch/secret/admin-key \
+          -XPUT 'https://localhost:9200/_cluster/settings' \
+          -d '{ "transient": { "cluster.routing.allocation.enable" : "none" } }'
+
+1. Perform a shard sync'd flush to ensure there are no pending operations waiting to be
+written to disk prior to shutting down.
+From any ES pod in the cluster:
+
+  curl -s --cacert /etc/elasticsearch/secret/admin-ca \
+          --cert /etc/elasticsearch/secret/admin-cert \
+          --key /etc/elasticsearch/secret/admin-key \
+          -XPUT 'https://localhost:9200/_flush/synced'
+
+1. Waiting for completion between each one, for each DC you have for an ES cluster,
+use `oc rollout latest` to deploy the latest version of the DC object. You will see
+a new pod deployed. Once the pod has two ready containers, you can move on to the
+next DC.
+1. Once all DCs for the cluster have been rolled out, re-enable shard balancing.
+Again from any ES pod in the cluster:
+
+  curl -s --cacert /etc/elasticsearch/secret/admin-ca \
+          --cert /etc/elasticsearch/secret/admin-cert \
+          --key /etc/elasticsearch/secret/admin-key \
+          -XPUT 'https://localhost:9200/_cluster/settings' \
+          -d '{ "transient": { "cluster.routing.allocation.enable" : "all" } }'
+
+1. Enable all external communications to the ES cluster now that restarting is complete.
```

Comment:
Once the restart is complete, enable all external communications to the ES cluster.